pkg/lightning/backend/local/region_job.go

```diff
@@ -973,6 +978,85 @@ type regionJobRetryer struct {
 	cancel context.CancelFunc
 }
 
+type dispatcher struct {
+	workerCtx context.Context
+
+	jobFromWorkerCh chan *regionJob
+	jobWg           *sync.WaitGroup
+
+	retryer *regionJobRetryer
+}
+
+func newDispatcher(
+	workerCtx context.Context,
+	jobFromWorkerCh chan *regionJob,
+	jobWg *sync.WaitGroup,
+	retryer *regionJobRetryer,
+) *dispatcher {
+	return &dispatcher{
+		workerCtx:       workerCtx,
+		jobFromWorkerCh: jobFromWorkerCh,
+		jobWg:           jobWg,
+		retryer:         retryer,
+	}
+}
+
+func (d *dispatcher) run() error {
```

Comment:
what's done in this loop is more like a post-processing of the job which finished one round of running, we can make this part of logic run directly inside job worker, not sure how much performance gain we get from pipeline execution, this part of logic is quite simple.
but anyway, no need to change in this pr